x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForNoFollowersStep.java

```diff
@@ -49,10 +53,31 @@ public boolean isRetryable() {
 
     @Override
     public void evaluateCondition(Metadata metadata, Index index, Listener listener, TimeValue masterTimeout) {
+        XPackInfoRequest xPackInfoRequest = new XPackInfoRequest();
```

Comment:
We have a bit of precedent here as we already skip the `UnfollowAction` completely if no CCR information is present in the index metadata.
See https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/UnfollowAction.java#L62 

How about we do the same here? (we have the `Metadata` and the `Index` so we can use a similar check here and if the `ccr` metadata is missing  for this index we notify `listener.onResponse(true)`) 

This avoids an extra API transport call.

Reply:
@andreidan As i read the code,  
`UnfollowAction` just check in the following cluster, and  there is a `ccr` KEY  in the metadata 
BUT
`WaitforNoFollowersStep` check in the leader cluster to ensure policy run safely that all followers stoped, and introduced in #38140.  There is no `ccr` KEY in metadata statement at leader cluster. so we can not do the same as `UnfollowAction`. 

The leader cluster get the CCR status only from the shards stats in `CCR_LEASE_KEY` AND feature API which CCR is disabled

Reply:
> > There is no ccr KEY in metadata statement at leader cluster
> 
> Ah of course ! Good shout @luyuncheng

@andreidan  i think `XPackInfoRequest` is a more lightweight calling VS `shards stats` requests. It can reduce master AND data nodes calculating, memory pressure without collect shards stats

Reply:
++ I think it makes sense, thanks for looking into this